src/plugins/flow.js

```diff
@@ -510,7 +523,8 @@ pp.flowParsePrimaryType = function () {
       return this.flowIdentToTypeAnnotation(startPos, startLoc, node, this.parseIdentifier());
 
     case tt.braceL:
-      return this.flowParseObjectType();
+    case tt.braceBarL:
+      return this.flowParseObjectType(false, true);
```

Comment:
If the token === tt.braceL shouldn't the call be `this.flowParseObjectType(false, false);`. It probably doesn't really change anything but it would be more clear maybe (and saves us one comparison in the `flowParseObjectType`, although it probably doesn't have an impact.)